src/App.js

```diff
@@ -14,11 +14,85 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export class App extends React.Component {
+  state = {
+    buttonVisible: true,
+    basketOfProducts: null,
+  }
+
+  addProduct = (product) => {
+    const basket = this.state.basketOfProducts;
+
+    if (basket !== null && basket.length >= 1) {
```

Comment:
```suggestion
    if (basket !== null) {
```
Это условие лишнее, у тебя `basket !== null`, только если его длина > 0